### PR TITLE
Remove AIP-44 from airflow/sensors/base.py (2nd Attempt)

### PR DESCRIPTION
**Pull request by jscheffl**
related: https://github.com/apache/airflow/issues/44436

Second attempt of #44518, hoping to NOT hit mypy issues again in all providers. Implemented it as it was prior Internal API w/o touching the sensor interface.

- [x] ./airflow/sensors/base.py: _orig_start_date

### CODE REVIEW

**Comment by jscheffl** [comment]
Okay, mypy checks again, I'll shortly wait until #44434 is merged, then these problems will be gone with a re-base

**Review by gopidesupavan — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**
